hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,243 @@
+/*
+Copyright 2024 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+)
+
+type ShortSummary struct {
+	NumberOfTests int
+	NumberOfFail  int
+	NumberOfPass  int
+	NumberOfSkip  int
+	FailedTests   []string
+	PassedTests   []string
+	SkippedTests  []string
+	Durations     map[string]float64
+	TotalDuration float64
+	GopoghVersion string
+	GopoghBuild   string
+	Detail        struct {
+		Name     string
+		Details  string
+		PR       string
+		RepoName string
+	}
+}
+
+// ParseEnvironmentList read the existing environments from the file
+func ParseEnvironmentList(listFile string) ([]string, error) {
+	data, err := os.ReadFile(listFile)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(string(data)), "\n"), nil
+}
+
+func GetTestSummariesFromGcp(pr, rootJob string, envList []string, client *storage.Client) (map[string]*ShortSummary, error) {
+	envToSummaries := map[string]*ShortSummary{}
+	for _, env := range envList {
+		if summary, err := getTestSummaryFromGCP(pr, rootJob, env, client); err == nil {
+			if summary != nil {
+				// if the summary is nil(missing) we just skip it
+				envToSummaries[env] = summary
+			}
+		} else {
+			return nil, fmt.Errorf("failed to fetch %s test summary from gcp, err: %v", env, err)
+		}
+	}
+	return envToSummaries, nil
+}
+
+// getFromSummary get the summary of a test on the specified env from the specified summary.
+func getTestSummaryFromGCP(pr, rootJob, env string, client *storage.Client) (*ShortSummary, error) {
+	ctx := context.TODO()
+
+	btk := client.Bucket("minikube-builds")
+	obj := btk.Object(fmt.Sprintf("logs/%s/%s/%s_summary.json", pr, rootJob, env))
+
+	reader, err := obj.NewReader(ctx)
+	if err != nil {
+		if err == storage.ErrObjectNotExist {
+			// if this file does not exist, just skip it
+			return nil, nil
+		}
+		return nil, err
+	}
```

Comment:
```suggestion
	if errors.Is(err, storage.ErrObjectNotExist) {
		// if this file does not exist, just skip it
		return nil, nil
	}
	if err != nil {
		return nil, err
	}
```
Unnest check since they can be checked independently

Also use `errors.Is` since it supports unwrapping